lib/diyepw/diyepw/meteorology.py

```diff
@@ -0,0 +1,263 @@
+
+        ############################
+        # Read TMY3 header
+        ############################
+        instance._headers = []
+        for i in range(0, 8):
+            line = tmy3_file.readline().strip()
+            instance._headers.append(line)
+        tmy3_file.close()
+
+        first_line = instance._headers[0].split(',')
+        instance._city = first_line[1]
+        instance._state = first_line[2]
+        instance._country = first_line[3]
+        instance._station_number = first_line[5]
+        instance._latitude = float(first_line[6])
+        instance._longitude = float(first_line[7])
+        instance._timezone_gmt_offset = float(first_line[8])
+        instance._elevation = float(first_line[9])
+
+        instance._comment = instance._headers[5]
+
+        ############################
+        # Read TMY3 data
+        ############################
+        data = np.genfromtxt(file_path, delimiter=',', skip_header=8)
+        instance._observations = pd.DataFrame(data={
+            "year":           [ int(i) for i in data[:, 0] ],
+            "month":          [ int(i) for i in data[:, 1] ],
+            "day":            [ int(i) for i in data[:, 2] ],
+            "hour":           [ int(i) for i in data[:, 3] ],
+            "minute":         [ int(i) for i in data[:, 4] ],
+            "Tdb":            data[:, 6],
+            "Tdew":           data[:, 7],
+            "RH":             data[:, 8],
+            "Patm":           data[:, 9],
+            "ExHorRad":       data[:, 10],
+            "ExDirNormRad":   data[:, 11],
+            "HorIR":          data[:, 12],
+            "GHRad":          data[:, 13],
+            "DNRad":          data[:, 14],
+            "DHRad":          data[:, 15],
+            "GHIll":          data[:, 16],
+            "DNIll":          data[:, 17],
+            "DHIll":          data[:, 18],
+            "ZenLum":         data[:, 19],
+            "Wdir":           data[:, 20],
+            "Wspeed":         data[:, 21],
+            "TotSkyCover":    data[:, 22],
+            "OpSkyCover":     data[:, 23],
+            "Visib":          data[:, 24],
+            "CeilH":          data[:, 25],
+            "PresWeathObs":   data[:, 26],
+            "PresWeathCodes": data[:, 27],
+            "PrecWater":      data[:, 28],
+            "AerOptDepth":    data[:, 29],
+            "SnowDepth":      data[:, 30],
+            "DSLS":           data[:, 31],
+            "Albedo":         data[:, 32],
+            "LiqPrecDepth":   data[:, 33],
+            "LiqPrecQuant":   data[:, 34]
+        })
+
+        return instance
+
+    def validate_against_epw_rules(self) -> list:
+        """
+        Check all observations to see whether they violate any restrictions that would prevent them
+        from being used in an EPW file
+
+        :return: A list of validation errors
+        """
+        violations = []
+
+        for col_name in _RANGES:
```

Comment:
once `_RANGES` is inside the class, you can call several ways, one of which is `Meterology._RANGES`